Pulling back new changes from deployment

viktorcsimma opened the pull request:
This is to merge the improvements @horcsinbalint has made to the development branch.

@kdmnk what do you think? ~~I can review it, but maybe only on Sunday or after that.~~ Oh, the system doesn't let me do it; since it's technically my PR... but I can add comments, so it could be solved.

horcsinbalint committed:
Update TrustProxies.php

horcsinbalint committed:
Temporarly disable printer

horcsinbalint committed:
Added additional parameters for printers

horcsinbalint committed:
Added debug message

horcsinbalint committed:
Fixed order

horcsinbalint committed:
Reenable printer

horcsinbalint committed:
Added arguments for printer stats

horcsinbalint committed:
Use config instead of env for production #364

horcsinbalint committed:
Fix bugs

horcsinbalint committed:
Fix bugs

horcsinbalint committed:
.

horcsinbalint committed:
Fix typo

horcsinbalint committed:
Add logging

horcsinbalint committed:
Add logging

horcsinbalint committed:
Fix pdflatex path

kdmnk commented on app/Console/Commands.php:
```diff
@@ -83,9 +83,10 @@ public static function latexToPdf($path, $outputDir)
             $result = "ok";
         } else {
-            $command = "pdflatex " . "-interaction=nonstopmode -output-dir " . $outputDir . " " . $path . " 2>&1";
+            $command = "/usr/bin/pdflatex " . "-interaction=nonstopmode -output-dir " . $outputDir . " " . $path . " 2>&1";
```
add the full path to the config then

kdmnk commented on app/Http/Middleware/TrustProxies.php:
```diff
@@ -12,7 +12,7 @@ class TrustProxies extends Middleware
      * @var array|string
      */
-    protected $proxies;
+    protected $proxies = ['172.5.0.10'];
```
What's this and why?

horcsinbalint replied:
We are running mars behind a reverse proxy and setting the reverse proxy as a trusted proxy for Laravel is mandatory if we want to use HTTPS. (https://laravel.com/docs/9.x/requests#configuring-trusted-proxies) There are some other hacky ways of setting it up but I didn't like them.
I would like it to be configurable from the .env, and I will add this to the PR.

kdmnk replied:
Okay, and add a comment here. Will this affect local development?

horcsinbalint replied:
I'm not 100% sure, and have not tested it, but I think it only comes in to effect if at least one of the headers in the request:
```
Request::HEADER_X_FORWARDED_FOR |
Request::HEADER_X_FORWARDED_HOST |
Request::HEADER_X_FORWARDED_PORT |
Request::HEADER_X_FORWARDED_PROTO |
Request::HEADER_X_FORWARDED_AWS_ELB
```
Which are mostly stuff that can come from a reverse proxy / load-balancer. But I think it should not have any effect unless you are connecting from IP `172.5.0.10` and the `APP_URL`'s protocol is https.

I will look into it today or tomorrow, but I think it should not have any effect for local development.

horcsinbalint replied:
I have tested it in my local environment and have not found any issues. So I think it should be fine

BertalanD replied:
Could we make this configurable instead of hard-coding the IP in the source code?

horcsinbalint replied:
Yep, I would like to make it configurable, I just currently have a long lead-time

horcsinbalint replied:
Or if someone would fix it, I would happily give advice and / or check the code

horcsinbalint commented:
Unfortunately I'm currently quite busy with my other tasks and other issues with the server. I'm planning on fixing the issues with this PR in about 3-4 weeks.

BertalanD commented on .env.example:
```diff
@@ -68,6 +68,8 @@ PREVENT_ACCESSING_MISSING_ATTRIBUTES=false
 PRINT_COST_ONESIDED=8
 PRINT_COST_TWOSIDED=12
 PRINTER_NAME=NemUjBela
+PRINTER_ADDITIONAL_ARGS=
+PRINTER_STAT_ADDITIONAL_ARGS=
```
I'm not sure how much of the changes to printing will be necessary once the printer server is redesigned (eotvoscollegium/smaugreinstall#8). I'm planning on tackling that task on Thursday; could we delay this part until then?

horcsinbalint committed:
Change proxy IP

horcsinbalint committed:
Change proxy IP

horcsinbalint committed:
Added config for commands (#400)

* Added config for commands

* style: format code with PHP CS Fixer

This commit fixes the style issues introduced in f1d870d according to the output
from PHP CS Fixer.

Details: https://github.com/EotvosCollegium/mars/pull/400

* Fixed a bug and added some comments

* style: format code with PHP CS Fixer

This commit fixes the style issues introduced in 6e41962 according to the output
from PHP CS Fixer.

Details: https://github.com/EotvosCollegium/mars/pull/400

* Fixed command name

* style: format code with PHP CS Fixer

This commit fixes the style issues introduced in 6dfdd35 according to the output
from PHP CS Fixer.

Details: https://github.com/EotvosCollegium/mars/pull/400

---------

Co-authored-by: deepsource-autofix[bot] <62050782+deepsource-autofix[bot]@users.noreply.github.com>

horcsinbalint committed:
Add config options (#401)

* Added config for commands

* style: format code with PHP CS Fixer

This commit fixes the style issues introduced in f1d870d according to the output
from PHP CS Fixer.

Details: https://github.com/EotvosCollegium/mars/pull/400

* Fixed a bug and added some comments

* style: format code with PHP CS Fixer

This commit fixes the style issues introduced in 6e41962 according to the output
from PHP CS Fixer.

Details: https://github.com/EotvosCollegium/mars/pull/400

* Fixed command name

* fix typo

---------

Co-authored-by: deepsource-autofix[bot] <62050782+deepsource-autofix[bot]@users.noreply.github.com>

horcsinbalint commented:
I think I have fixed the issues that have been pointed out. @kdmnk @viktorcsimma Could you please check it?

Should fix #364 #366 #372 #373

kdmnk committed:
Merge branch 'development' into deployment
